### PR TITLE
Realized layer dao with methods findeByLogin, createUser and connecti…

### PR DESCRIPTION
**Pull request by Fleaman777**
New layer dao with method findeByLogin: сhecks if there is such a user in the database, 
createUser: adds new user in data base,
getConnectional: gives a new connection from the pool to connect to the database.

### CODE REVIEW

File: src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import javax.naming.NamingException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.List;
+
+public class ConnectionPool {
+
+    static List<Connection> connectionPool = new LinkedList<Connection>();
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java]: Please, use access modifiers

File: src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    Connection connect = null;
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java]: Please, use access modifiers

File: src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            connectReturn = connectionPool.remove(0);
+        }
+        return connectReturn;
+    }
+
+    public static void close(Connection connection) {
+        if(connection != null) {
+            connectionPool.add(connection);
+        }
+        else connectionPool.add(new ConnectionPool().connect);
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java]: It looks strange and unclear. I would recommend to use another approach:
```
ConnectionPool pool = new ConnectionPool();
Connection con = pool.getConnection();
connectionPool.add(con);
```
Also please do not forget about `{}` brackets in `if-else` statements

File: src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    private static void addConnect() {
+        for (int i = 1; i <= 10; i++) {
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java]: 10 is a magic number. Please, use constants

File: src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            e.printStackTrace();
+            System.exit(1);
+        }
+        System.out.println("PostgreSQL JDBC Driver Registered!");
+
+
+        try {
+
+            connect = DriverManager.getConnection(
+                    "jdbc:postgresql://127.0.0.1:5432/payment_system");
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java]: Please, move it to constants

File: src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        if (connectionPool.size() < 1) {
+            addConnect();
+            connectReturn = connectionPool.remove(0);
+        } else {
+            connectReturn = connectionPool.remove(0);
+        }
+        return connectReturn;
+    }
+
+    public static void close(Connection connection) {
```
Comment (nikandtos) [inline src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java]: Please, find a decent name for method.

File: src/main/java/com/epam/lab/paymentsystem/dao/UserDAO.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+                e.printStackTrace();
+            }
+            PreparedStatement ps = connect.prepareStatement(sql);
+            ps.setString(1, user.getLogin());
+            ps.setString(2, user.getPassword());
+            ps.setString(3, user.getName());
+
+            ps.executeUpdate();
+            ResultSet rs = ps.executeQuery();
+            user.setId(rs.getInt(1));
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/dao/UserDAO.java]: You should not change the incoming data, you have to create a new user and fill it with proper values. You might as well create additional constructor\clone method for copying data. For example:
```
public User(User oldUser) {
   login = oldUser.getLogin();
   ... etc
}

OR

public static User getCopy(User old) {
   User newUser = new User();
   newUser.setLogin(old.getLogin);
   ... etc
   return newUser;
}
```

File: src/main/java/com/epam/lab/paymentsystem/dao/UserDAO.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+import java.sql.*;
+import com.epam.lab.paymentsystem.entities.User;
+
+
+public class UserDAO implements UserDAOInterface {
+
+
+    public User createUser(User user) {
+
+        String sql = "INSERT INTO users (login, password, user_name)" + "VALUES (?, ?, ?)";
```
Comment (nikandtos) [inline src/main/java/com/epam/lab/paymentsystem/dao/UserDAO.java]: All queries should be constants.

File: src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    private static void addConnect() {
+        for (int i = 1; i <= 10; i++) {
+            connectionPool.add(new ConnectionPool().connect);
+        }
+    }
+
+    private ConnectionPool() {
+
+        try {
+            Class.forName("dao.UserDAO");
```
Comment (Meisssshu) [inline src/main/java/com/epam/lab/paymentsystem/dao/ConnectionPool.java]: I think it is better to user UserDAO.class.getName() or smth like that

**Pull request merged**